### PR TITLE
Redesign WebView2 TOC & clarify article titles

### PR DESCRIPTION
**Pull request by mikehoffms**
Rendered articles for review:
* **WebView2 documentation** (landing page)
   * `/webview2/landing/index.yml`
   * https://review.learn.microsoft.com/microsoft-edge/webview2/landing/?branch=pr-en-us-3124
      * To load latest .png files & TOC: **F12** > right-click **Refresh** > **Empty cache and hard refresh** > **F12**.
   * [Before](https://learn.microsoft.com/microsoft-edge/webview2/)
   * Added new landing page.
* **TOC**
   * `toc.yml`
      * To load latest .png files & TOC: **F12** > right-click **Refresh** > **Empty cache and hard refresh** > **F12**.
   * Removed abstract TOC buckets ("Fundamentals", "Advanced") and directly/immediately shows their content.
      * Removed unnecessary levels of nesting.
      * Created "Introduction and overview" toc bucket.
      * Used tech-topic TOC bucket names.  Promoted the tech key words users are looking for, and a generally flatter design that reveals more articles and specific topics by default.
      * Enabled more flexible grouping/placement of the articles.
      * Eliminated top-level buckets that are low-value & abstract; promotes the high-value, keyword-based nodes that were contained in them.
      * Eliminated a UX of guessing and having to explore into the abstract TOC nodes.
      * Made maintenance/improvement easier.
   * Group articles by tech topic (high-value grouping), not by some other, low-value quality such as info type or Fundamentals vs. Advanced.
      * Grouped together articles such as `distribution.md` & `versioning.md` that were spread apart between two different TOC buckets (Fundamentals & Deployment).
      * Within a tech-topic toc bucket, basic articles are first; advanced or specialized articles are last.
      * Instead of "Fund/Adv" buckets, an article's Intro helps user decide if article is relevant.
   * Improved titles of articles (globally in repo):
      * Changed "Sample apps" to "WebView2 sample apps".
      * Changed "Overview of WebView2 features and APIs" to "Overview of WebView2 APIs".
      * Changed "Working with local content in WebView2 apps" to "Using local content in WebView2 apps".
      * Changed "Interop of native-side and web-side code" to "Interop of native and web code".
      * Changed "Customizing the UI in WebView2 apps" to "Customizing the UI of WebView2 apps".
      * Changed "Deployment samples" to "WebView2 deployment samples".
* **Prerelease and Release SDKs for WebView2**
   * https://review.learn.microsoft.com/microsoft-edge/webview2/concepts/versioning?branch=pr-en-us-3124
   * [Before/Live](https://learn.microsoft.com/microsoft-edge/webview2/concepts/versioning)
   * Changed article title from: Understand the different WebView2 SDK versions.
* **Interop of native and web code**
   * https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/communicate-btwn-web-native?branch=pr-en-us-3124
   * [Before/Live](https://learn.microsoft.com/microsoft-edge/webview2/how-to/communicate-btwn-web-native)
   * Changed article title from: Interop of native-side and web-side code

#### Mapping from existing TOC to redesigned TOC

image

#### History of PRs
This PR 3124 replaces forked July 2023 PR https://github.com/MicrosoftDocs/edge-developer/pull/2725, which lists the irrelevant file archive.md as entirely changed by that PR, yet not at all changed by that PR.

AB#50063950

### CODE REVIEW

File: microsoft-edge/toc.yml [hunk 1]
```diff
@@ -1209,8 +1205,9 @@
         - name: Overview of WebView2 features and APIs
           href: webview2/concepts/overview-features-apis.md
 
-# -----------------------------------------------------------------------------
-# machine setup
+        - name: Differences between Microsoft Edge and WebView2
+          href: webview2/concepts/browser-features.md
+
         - name: Set up your Dev environment for WebView2
           href: webview2/how-to/machine-setup.md
```
Comment (mikehoffms) [inline microsoft-edge/toc.yml:1212]:
```suggestion
        - name: Set up your Dev environment for WebView2
          href: webview2/how-to/machine-setup.md

        - name: Differences between Microsoft Edge and WebView2
          href: webview2/concepts/browser-features.md
```

File: microsoft-edge/developer/index.yml
Comment (mikehoffms) [inline microsoft-edge/developer/index.yml:1]: **todo:** Recommend using articles in this sequence, b/c fastest & easiest w/ greatest guarantee of success:
1. Install/run main sample.
2. Install/run sample for your target platform. 
3. Install/run finished GetStart for your target platform.
4. Do GetStart steps to add wv2 code to VS project for your platform.

Comment (mikehoffms) [inline microsoft-edge/developer/index.yml:1]: Create a separate PR to propose deleting wv2's debug.md.
* That used to contain single tabset containing 3 separate articles, which became 3 .md's.
* No article links to here.

Propagate parts of its lead-in to top of the 3 child articles.

Comment (mikehoffms) [inline microsoft-edge/developer/index.yml:1]: Create a separate PR to propose deleting Roadmap.md

Comment (mikehoffms) [inline microsoft-edge/developer/index.yml:1]: i'll make sure no article was dropped (a periodic check)

**Review by mikehoffms — COMMENTED** [review]
LGTM - Content Developer/Writer/Editor

**Review by sivMSFT — APPROVED** [review]
LGTM - PM

**Pull request merged**